src/groupeddataframe/groupeddataframe.jl

```diff
@@ -214,15 +221,15 @@ function Base.getindex(gd::GroupedDataFrame, idxs::AbstractVector{<:Integer})
             new_groups[idx[j]] = i
         end
     end
-    GroupedDataFrame(gd.parent, gd.cols, new_groups, gd.idx,
+    GroupedDataFrame(gd.parent, copy(gd.cols), new_groups, gd.idx,
                      new_starts, new_ends, length(new_starts), nothing,
                      Threads.ReentrantLock())
 end
 
 # Index with colon (creates copy)
 function Base.getindex(gd::GroupedDataFrame, idxs::Colon)
     Threads.lock(gd.lazy_lock)
-    new_gd = GroupedDataFrame(gd.parent, gd.cols, gd.groups, getfield(gd, :idx),
+    new_gd = GroupedDataFrame(gd.parent, copy(gd.cols), gd.groups, getfield(gd, :idx),
```

Comment:
I preferred to copy it, because it is cheap (`gd.cols` is a very short vector) and I felt it was safer to "`copy` unless it hits performance". But I can remove this copy if you prefer.

Reply:
Not a big deal, but given that we don't copy any other field we're very unsafe anyway.

Reply:
Agreed - but I would leave it as is here as it is cheap (I agree that in general we are very unsafe here in general).